Adyen/Core/APIClient/AnyAPIContext.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) 2021 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+
+/// :nodoc:
+/// An object that needs an API context to retrieve internal resources
+public protocol APIContextAware: AnyObject {
+    
+    /// :nodoc:
+    /// The API context
+    var apiContext: AnyAPIContext { get }
+    
+}
+
+/// :nodoc:
+/// An API context that defines parameters for retrieving internal resources
+public protocol AnyAPIContext {
+    
+    /// :nodoc:
+    /// The environment to retrieve internal resources from.
+    var environment: AnyAPIEnvironment { get }
+    
+    /// :nodoc:
+    /// The HTTP headers.
+    var headers: [String: String] { get }
+    
+    /// :nodoc:
+    /// The query parameters.
+    var queryParameters: [URLQueryItem] { get }
+    
+    /// :nodoc:
+    /// The client key.
+    var clientKey: String { get }
```

Comment:
I believe `clientKey ` supposed to be part of `APIContext`, not `AnyAPIContext` ?

Reply:
I believe so as well, clientKey is an implementation details of the specific `APIContext`

Reply:
I believe so as well